health/service.go

```diff
@@ -0,0 +1,151 @@
+package health
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+)
+
+// CheckPath is the baked in SOH endpoint path.
+const CheckPath = "/soh"
+
+// Service provides a mechanism to update a service SOH status.
+type Service struct {
+	mu sync.Mutex
+
+	// status is used to indicate whether the service is running
+	status bool
+	// last stores the time of the last update.
+	last time.Time
+
+	// start stores when the service was started.
+	start time.Time
+	// aged is the time if no updates have happened indicates the service is no longer running.
+	// set to 0 if no age check needed
+	aged time.Duration
+	// startup is the time after the start which the check is assumed to be successful.
+	startup time.Duration
+}
+
+// New returns a health Service which provides running SOH capabilities.
+func New(endpoint string, aged, startup time.Duration) *Service {
+	service := &Service{
+		aged:    aged,
+		last:    time.Now(),
+		start:   time.Now(),
+		startup: startup,
+	}
+
+	router := http.NewServeMux()
+	router.HandleFunc(CheckPath, service.handler)
+
+	srv := &http.Server{
+		Addr:              endpoint,
+		Handler:           router,
+		ReadHeaderTimeout: 2 * time.Second,
+	}
+
+	go func() {
+		_ = srv.ListenAndServe()
```

Comment:
Should check the error, return if not nil.

Reply:
it is in a separate routine, if the service is not started, the health check would fail anyway, but should log the error

Reply:
What happens when the health check fails?

Reply:
ECS will mark it as `UNHEALTHY` and the service is restarted